.github/actions/check-docsite-class-hashes/index.js

```diff
@@ -0,0 +1,53 @@
+import fs from "fs/promises";
+import * as core from "@actions/core";
+
+const classHashesFilePath = "docs/modules/ROOT/pages/utils/_class_hashes.adoc";
+
+async function main() {
+  const compilerVersionInput = core.getInput("compiler-version");
+  const jsonFileInput = core.getInput("json-file");
+
+  let content = await fs.readFile(jsonFileInput, 'utf8');
+
+  // Remove the first 4 lines
+  // TODO: Fix the output of the class hashes to not include the first 4 lines
+  let json = content.split('\n').slice(4).join('\n');
+  let hashes = JSON.parse(json);
+
+  let expectedContent = createHashesFileNewContent(hashes.contracts, compilerVersionInput);
+  let foundContent = await fs.readFile(classHashesFilePath, 'utf8');
+
+  if (expectedContent !== foundContent) {
+    core.debug(expectedContent);
+    core.debug(foundContent);
+    core.setFailed("Class hashes file is not up to date");
+  }
+}
+
+function createHashesFileNewContent(contracts, cairoVersion) {
+    const header = `// Version\n:class-hash-cairo-version: \
+https://crates.io/crates/cairo-lang-compiler/${cairoVersion}[cairo ${cairoVersion}]`;
+
+    let hashes = "// Class Hashes\n";
+    // The slice 13 is to remove the "openzeppelin_" prefix
+    hashes += contracts.sort(compareContracts).map(contract => {
+        return `:${contract.name.slice(13)}-class-hash: ${contract.sierra}`;
+    }).join('\n');
```

Comment:
I wonder if it'd be easier to add a command to `starknet_class_hash` that outputs a stripped down version of the hashes and then we could just compare it to `:x-class-hash:` in _class_hashes.adoc. All the work is basically done with the current approach, so maybe it's not worth it. Just a thought